components/salsa-macro-rules/src/setup_tracked_fn.rs

```diff
@@ -171,7 +180,7 @@ macro_rules! setup_tracked_fn {
                         let zalsa = db.zalsa();
                         $INTERN_CACHE.get_or_create(zalsa, || {
                             <dyn $Db as $Db>::zalsa_register_downcaster(db);
-                            zalsa.add_or_lookup_jar_by_type::<$Configuration>().successor(0)
+                            zalsa.lookup_jar_by_type::<$Configuration>().get_or_create().successor(0)
```

Comment:
Nit: Should we use the same two-pased registration here to only register the downcasters in the create branch?